### ISSUE TITLE
sae_manifold_fit: default max_iter=50 / learning_rate=0.04 under-converges drastically — single periodic atom recovers R²=0.16 on 1-harmonic data, needs ~it=400 lr=0.2 to reach R²=0.99

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#180]
## Observed
On the exact 1-Fourier-harmonic synthetic from
`tests/test_sae_manifold_curved_beats_linear.py` (n=400, p=64), the
test calls `sae_manifold_fit` with `max_iter=50, learning_rate=0.04`
and gets `R²(curved K=1, d=2) ≈ 0.16`. The test asserts
`R² ≥ 0.9` and fails.

Sweeping `max_iter` × `learning_rate` on the same data (with the only
runnable basis/prior combo on CPU, `periodic + ibp_map` — see #168,
#169) shows the model is just badly under-converged at the test's
defaults:

| max_iter | learning_rate | R²(curved K=1 d=2) |
| -------- | ------------- | ------------------ |
| 50       | 0.04          | **0.16**           |
| 50       | 0.2           | 0.46               |
| 100      | 0.1           | 0.85               |
| 200      | 0.1           | **0.83**  (worse than it=100) |
| 200      | 0.2           | 0.92 — 0.98 (data-dependent) |
| 400      | 0.2           | 0.99               |

Two distinct problems:

1. **Convergence stalls below R²=0.5 at default `lr=0.04`** even for a
   one-atom periodic fit to a literal 1-harmonic mixture. The defaults
   should at minimum recover this trivial case.
2. **`R²` is non-monotonic in `max_iter`.** Going from 100 → 200 iters
   at `lr=0.1` *decreased* R² from 0.8492 to 0.8273 on the same input,
   same seed. A well-behaved (V)EM / coordinate-descent loop should
   never go backwards on the joint objective; this suggests the line
   search / Newton step / penalty schedule occasionally takes
   destabilising steps.

## Reproducer (≤30 lines)
```python
import math, numpy as np, gamfit
rng = np.random.default_rng(7)
n, p = 400, 64
theta = rng.uniform(0, 2*math.pi, n)
H = np.column_stack([np.cos(theta), np.sin(theta)])
mix = rng.normal(size=(2,p)); mix /= np.linalg.norm(mix,axis=0,keepdims=True)
z = H @ mix + 0.04 * rng.normal(size=(n,p))
z -= z.mean(0, keepdims=True)

def r2(x,f):
    ss=float(np.sum((x-f)**2)); tt=float(np.sum((x-x.mean(0))**2))
    return 1.0-ss/max(tt,1e-12)

for it, lr in [(50,0.04),(100,0.1),(200,0.1),(200,0.2),(400,0.2)]:
    r = gamfit.sae_manifold_fit(Z=z, n_atoms=1, atom_basis="periodic",
        atom_dim=2, assignment_prior="ibp_map",
        max_iter=it, learning_rate=lr, random_state=0)
    print(f"it={it} lr={lr} R2={r2(z, r.fitted):.4f}")
```

## Expected
- Either: defaults (`max_iter=50, learning_rate=0.04`) should be sized
  to converge on the trivial 1-harmonic K=1 d=2 case to ≥ 0.9.
- Or: the convergence loop should auto-terminate based on relative
  objective improvement / step length rather than a fixed iteration
  budget, so that the user does not have to find magic numbers (200 +
  0.2 vs. 400 + 0.2 vs. 200 + 0.1 each give very different answers on
  the same input).
- The objective should be monotonically improving across outer
  iterations; the 100→200 iter regression at the same lr/seed should
  not happen.

This is the underlying behaviour behind the failing test
`tests/test_sae_manifold_curved_beats_linear.py`.

## Environment
- gamfit 0.1.123 (pypi wheel)
- macOS arm64, CPU backend
- Python 3.13

**Comment by SauersML** [comment]
Fixed by `81d77b43` (auto-derive `learning_rate` from `assignment_kind`) + `6b247894` (route periodic `atom_dim` into `n_harmonics`, keep `latent_dim=1` so the analytic `PeriodicHarmonicEvaluator` stays live across Newton steps).

Together these make the default Python entry point select `lr=1.0` for softmax / ibp_map (the natural full Gauss-Newton step under Armijo backtracking) instead of the historical `lr=0.04` that was starving the assignment posterior of gradient mass and letting the IBP sigmoid drift into the saturated tail.

Regression coverage: `sae_manifold_fit_10_steps_one_harmonic_reaches_high_r2` in `src/terms/sae_manifold.rs:2921` — asserts in-sample R² ≥ 0.95 in 10 Newton steps on the 1-harmonic K=1 periodic case starting from a phase-shifted init at `lr=1.0` (which is what auto-lr now picks for softmax/ibp_map).

Closing as fixed.
